data_pipelines_cli/dbt_cloud_api_client.py

```diff
@@ -0,0 +1,224 @@
+        """
+        To find your user ID in dbt Cloud, read the following steps:
+        1. Go to Account Settings, Team, and then Users,
+        2. Select your user,
+        3. In the address bar, the number after /users is your user ID.
+        """
+
+        self.token = token
+        """You can find your User API token in the Profile page under the API Access label"""
+
+    def request(self, url, data):
+        headers = {
+            "Accept": "application/json",
+            "Content-Type": "application/json",
+            "Authorization": f"Token {self.token}"
+        }
+        response = requests.post(
+            url=url,
+            data=data,
+            headers=headers
+        )
+        res = json.loads(response.content)
+        if not res["status"]["is_success"]:
+            raise Exception(res["status"]["user_message"] + "\n" + res["data"])
+        return res
+
+    def create_project(self, name):
+        """
+        Note: the dbt_project_subdirectory is an optional field which allows you to point
+        dbt Cloud to a subdirectory that lives within the root folder of your target repository.
+        This means dbt Cloud will look for a dbt_project.yml file at that location.
+
+        :param name: Name of the project
+        :return: ID of created project
+        """
+        new_project = {
+            "id": None,
+            "account_id": self.account_id,
+            "name": name,
+            "dbt_project_subdirectory": None,
+            "connection_id": None,
+            "repository_id": None
+        }
+
+        new_project_data = json.dumps(new_project)
+
+        response = self.request(f"{self.host_url}/v3/accounts/{str(self.account_id)}/projects/", new_project_data)
+        return response["data"]["id"]
+
+    def create_repository(self, project_id, git_clone_url):
+        """
+        Note: After creating  a dbt Cloud repository's SSH key, you will need to add the generated key text as
+         a deploy key to the target repository. This gives dbt Cloud permissions to read / write in the repository.
+
+        :param git_clone_url: Repository remote url
+        :param project_id: ID of the project
+        :return: repository ID and deploy key
+        """
+        new_repository = {
+            "account_id": self.account_id,
+            "project_id": project_id,
+            "remote_url": git_clone_url,
+            "git_clone_strategy": "deploy_key",
+            "github_installation_id": None,
+            "token_str": None
+        }
+
+        new_repository_data = json.dumps(new_repository)
+
+        response = self.request(
+            f"{self.host_url}/v3/accounts/{str(self.account_id)}/projects/{str(project_id)}/repositories/",
+            new_repository_data)
+        return response["data"]["id"], response["data"]["deploy_key"]["public_key"]
+
+    def create_development_environment(self, project_id):
+        """
+        Create development environment
```

Comment:
Is there any big query connection assigned to the environment?

Reply:
Connection is assigned to the project, not the environment.
Environments encompass a collection of settings for how you want to run your dbt project. This includes:
dbt version
git branch
data location (target schema).